### ISSUE TITLE
Speed up import for rank 30 objects

### ISSUE THREAD

**Issue by lonvia** [komoot/photon#529]
When importing objects from Nominatim to Photon, the importer needs to read the address information from the `place_addressline` table for each object. For objects at rank 30 the information is not read for the object itself but from its parent object, see [NominatimConnector:179](https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java#L179)ff.

Import could be sped up quite a bit if the parent address was cached and reused between rank 30 objects with the same parent. I was thinking along the lines of sorting the [object query](https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java#L230) not only by geometry_sector but also by parent_place_id and then maintaining a one-item cache of the last parent address retrieved for a rank 30 object.

Just a rough estimate for the potential savings: there are 140 mio. rank 30 objects in a planet database but they have only 17 mio. unqiue parent_place_ids between them.

**Comment by alfmarcua** [comment]
For the queries of the following type:
```sql
SELECT p.name, p.class, p.type, p.rank_address
FROM placex p, place_addressline pa
WHERE p.place_id = pa.address_place_id and pa.place_id = $1
and pa.cached_rank_address > 4 and pa.address_place_id != $2 and pa.isaddress
ORDER BY rank_address desc, fromarea desc, distance asc, rank_search desc
```
In order to speed up the process, another suggestion is to group them in batch or in a single query. This could even be  included in the query performed in readEntireDatabase method, see [NominatimConnector:228](https://github.com/komoot/photon/blob/2bbce026b58b44d33388d9bca0fe1747c32a9933/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java#L228).

**Comment by alfmarcua** [comment]
Another approach could relay in avoid the query for the term of the parent place ID itself, see [NominatimConnector:189](https://github.com/komoot/photon/blob/2bbce026b58b44d33388d9bca0fe1747c32a9933/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java#L189). Maybe that data can be cached or can be get altoghether with the previous query. If those queries are avoided, it could save up about a 7% of the total query time.

**Comment by lonvia** [comment]
Resolved by #547.
